scripts/build_plugin.py

```diff
@@ -0,0 +1,484 @@
+            sg_connection,
+            Descriptor.CONFIG,
+            base_config_def,
+            fallback_roots=[bundle_cache_root],
+            resolve_latest=using_latest_config
+        )
+
+    logger.info("Resolved config %r" % cfg_descriptor)
+    return cfg_descriptor
+
+def _validate_manifest(source_path):
+    """
+    Validate that the manifest file is present and valid.
+
+    :param source_path: Source path to plugin
+    :return: parsed yaml content of manifest file
+    """
+    # check for source manifest file
+    manifest_path = os.path.join(source_path, "info.yml")
+    if not os.path.exists(manifest_path):
+        raise TankError("Cannot find plugin manifest '%s'" % manifest_path)
+
+    logger.debug("Reading %s" % manifest_path)
+    with open(manifest_path, "rt") as fh:
+        manifest_data = yaml.load(fh)
+
+    logger.debug("Validating manifest...")
+    for parameter in REQUIRED_MANIFEST_PARAMETERS:
+        if parameter not in manifest_data:
+            raise TankError(
+                "Required plugin manifest parameter '%s' missing in '%s'" % (parameter, manifest_path)
+            )
+
+    return manifest_data
+
+def _bake_manifest(manifest_data, cfg_descriptor, core_descriptor, plugin_root):
+    """
+    Bake the info.yml manifest into a python file.
+
+    :param manifest_data: info.yml manifest data
+    :param cfg_descriptor: descriptor object pointing at the config to use
+    :param core_descriptor: descriptor object pointing at core to use for bootstrap
+    :param plugin_root: Root path for plugin
+    """
+    # add entry point to our module to ensure multiple plugins can live side by side
+    module_name = "sgtk_plugin_%s" % manifest_data["entry_point"]
+    full_module_path = os.path.join(plugin_root, "python", module_name)
+    filesystem.ensure_folder_exists(full_module_path)
+
+    # write init.py
+    with open(os.path.join(full_module_path, "__init__.py"), "wt") as fh:
+        fh.write("# this file was auto generated.\n")
+        fh.write("from . import manifest\n")
+        fh.write("# end of file.\n")
+
+    # now bake out the manifest into code
+    params_path = os.path.join(full_module_path, "manifest.py")
+    with open(params_path, "wt") as fh:
+
+        fh.write("# this file was auto generated.\n")
+
+        fh.write("\nbase_configuration=\"%s\"\n\n" % cfg_descriptor.get_uri())
+
+        for (parameter, value) in manifest_data.iteritems():
+
+            if parameter == "base_configuration":
+                # configuration is processed separately
+                continue
+
+            if isinstance(value, str):
+               fh.write("%s=\"%s\"\n" % (parameter, value.replace("\"", "'")))
+            elif isinstance(value, int):
+                fh.write("%s=%d\n" % (parameter, value))
+            elif isinstance(value, bool):
+                fh.write("%s=%s\n" % (parameter, value))
+            else:
+                raise ValueError(
```

Comment:
how about keeping it *really* simple for the time being? we can always extend it later if the need arises!